flax/experimental/nnx/tests/nn/test_embed.py

```diff
@@ -12,31 +12,63 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import typing as tp
+
 import jax
 from absl.testing import parameterized
 from jax import numpy as jnp
+import numpy as np
 from numpy.testing import assert_array_equal
 
 from flax import linen
 from flax.experimental import nnx
 
+DType = np.dtype
+Dtype = tp.Any
+
+
+class SupportsDType(tp.Protocol):
+  @property
+  def dtype(self) -> DType:
+    ...
 
-class TestLinenConsistency(parameterized.TestCase):
 
+DTypeLike = tp.Union[
+  str,  # like 'float32', 'int32'
+  type[tp.Any],  # like np.float32, np.int32, float, int
+  np.dtype,  # like np.dtype('float32'), np.dtype('int32')
+  SupportsDType,  # like jnp.float32, jnp.int32
+]
+
+
+class TestLinenConsistency(parameterized.TestCase):
   @parameterized.product(
-      input_dtype = [jnp.int16, jnp.int32],
-      dtype = [jnp.float32, jnp.float16],
-      param_dtype = [jnp.float32, jnp.float16],
+    input_dtype=[jnp.int16, jnp.int32],
+    dtype=[jnp.float32, jnp.float16],
+    param_dtype=[jnp.float32, jnp.float16],
   )
-  def test_nnx_linen_equivalence(self, input_dtype, **kwargs):
+  def test_nnx_linen_equivalence(
+    self,
+    input_dtype: tp.Optional[DTypeLike],
+    dtype: tp.Optional[Dtype],
+    param_dtype: Dtype,
```

Comment:
Note: in the `test_nnx_linen_equivalence` test in `test_embed.py`, the `input_dtype` arg is of `DType` typing (with an upper case T, that's derived from Jax), whereas the `dtype` and `param_dtype` arg is of `Dtype` typing (with a lower case t, that's derived from NNX). You can see their aliases at the top of the file [here](https://github.com/google/flax/pull/3562/files#diff-ba0465aa3c924f608942a9b6a40cb876fdcc50c4084d8080aa4569c783c4a413R26-R27).

Should we standardize the typing annotation (both in spelling as well as definition)?

Reply:
standarization would be nice, can get reuse something from JAX? It feels like they should expose this so we can reuse it if possible.